main.tf

```diff
@@ -20,12 +20,13 @@ resource "aws_vpn_gateway" "default" {
 
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
-  count       = local.enabled && var.customer_gateway_ip_address != null ? 1 : 0
-  device_name = module.this.id
-  bgp_asn     = var.customer_gateway_bgp_asn
-  ip_address  = var.customer_gateway_ip_address
-  type        = "ipsec.1"
-  tags        = module.this.tags
+  count            = local.enabled && var.customer_gateway_ip_address != null ? 1 : 0
+  device_name      = module.this.id
+  bgp_asn          = var.customer_gateway_bgp_asn <= 2147483647 ? var.customer_gateway_bgp_asn : null
+  bgp_asn_extended = var.customer_gateway_bgp_asn > 2147483647 ? var.customer_gateway_bgp_asn : null
```

Comment:
One thing to confirm with this argument before we merge: Was this arg introduced in a specific version of the AWS provider? Does our `versions.tf` for this module require that version and higher? If not, we'll need to update our pin of the AWS provider. Please confirm that before merge.

Reply:
Hi @Gowiem the module at the moment pins version `>= 2.0` of the AWS provider, while this feature was introduced in version `5.53.0`: I did not update `versions.tf` because it did not reflect changes on other attributes (i.e., `device_name` which was introduced in `3.28.0`).

If you confirm this has to be updated, I'm going to pin `>=5.53.0` in `versions.tf` and update the PR.

Reply:
@riccardolocci yes, please update the pin. Thanks!